spec/requests/api/blueprints_spec.rb

```diff
@@ -0,0 +1,196 @@
+RSpec.describe "Blueprints API" do
+  describe "GET /api/blueprints" do
+    it "lists all the blueprints with an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize collection_action_identifier(:blueprints, :read, :get)
+
+      run_get(blueprints_url)
+
+      expected = {
+        "count"     => 1,
+        "subcount"  => 1,
+        "name"      => "blueprints",
+        "resources" => [
+          hash_including("href" => a_string_matching(blueprints_url(blueprint.id)))
+        ]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "forbids access to blueprints without an appropriate role" do
+      FactoryGirl.create(:blueprint)
+      api_basic_authorize
+
+      run_get(blueprints_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe "GET /api/blueprints/:id" do
+    it "will show a blueprint with an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize action_identifier(:blueprints, :read, :resource_actions, :get)
+
+      run_get(blueprints_url(blueprint.id))
+
+      expect(response.parsed_body).to include("href" => a_string_matching(blueprints_url(blueprint.id)))
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "forbids access to a blueprint without an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize
+
+      run_get(blueprints_url(blueprint.id))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe "POST /api/blueprints" do
+    it "can create a blueprint" do
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(blueprints_url, :name => "foo", :description => "bar", :ui_properties => {:some => {:json => "baz"}})
```

Comment:
In SSUI Designer there currently isn't a way for a user to enter a description of the blueprint.